sys/targets/targets.go

```diff
@@ -764,6 +764,7 @@ func initTarget(target *Target, OS, arch string) {
 	if runtime.GOOS != target.BuildOS {
 		// Spoil native binaries if they are not usable, so that nobody tries to use them later.
 		target.CCompiler = fmt.Sprintf("cant-build-%v-on-%v", target.OS, runtime.GOOS)
+		target.BrokenCompiler = target.CCompiler
```

Comment:
It should be the reason why the compiler is broken, not the compiler binary name.

We use BrokenCompiler to silently skip tests rather than fail. Somehow we get this far w/o setting it here. I am a bit worried that this may lead to silently ignored bugs in future. E.g. we don't ever set it on CI.
Why do we need this now?

Reply:
We need this to address broken tests on OpenBSD/NetBSD:
https://github.com/google/syzkaller/pull/6280#issuecomment-3355516405

https://syzkaller.appspot.com/text?tag=CrashLog&x=109bd534580000

We could run `TestBuildDescriptions` on Linux only, but then the question whether it's okay at all to rely on BrokenCompiler in tests, which we do quite intensively.